tools/isobuild/compiler-plugin.js

```diff
@@ -679,100 +712,72 @@ class ResourceSlot {
     return isInImports;
   }
 
-  addStylesheet(options) {
-    const self = this;
-    if (! self.sourceProcessor) {
+  addStylesheet(options, lazyFinalizer) {
+    if (! this.sourceProcessor) {
       throw Error("addStylesheet on non-source ResourceSlot?");
     }
 
-    const data = files.convertToStandardLineEndings(options.data);
-    const useMeteorInstall = self.packageSourceBatch.useMeteorInstall;
-    const sourcePath = this.inputResource.path;
-    const targetPath = options.path || sourcePath;
-    const resource = {
-      refreshable: true,
-      sourcePath,
-      targetPath,
-      servePath: self.packageSourceBatch.unibuild.pkg._getServePath(targetPath),
-      hash: sha1(data),
-      lazy: this._isLazy(options, false),
-    };
+    // In contrast to addJavaScript, CSS resources passed to addStylesheet
+    // default to being eager (non-lazy).
+    options.lazy = this._isLazy(options, false);
 
-    if (useMeteorInstall && resource.lazy) {
+    if (this.packageSourceBatch.useMeteorInstall &&
+        options.lazy) {
       // If the current packageSourceBatch supports modules, and this CSS
       // file is lazy, add it as a lazy JS module instead of adding it
       // unconditionally as a CSS resource, so that it can be imported
       // when needed.
-      resource.type = "js";
-      resource.data =
-        Buffer.from(cssToCommonJS(data, resource.hash), "utf8");
+      this.addJavaScript(options, async () => {
+        const result = typeof lazyFinalizer === "function"
+          ? await lazyFinalizer()
+          : { data: options.data };
 
-      self.jsOutputResources.push(resource);
+        if (result) {
+          result.data = Buffer.from(cssToCommonJS(result.data), "utf8");
+        }
```

Comment:
This demonstrates how `lazyFinalizer` functions can be chained using normal function composition.